### PR TITLE
Add and use custom permissions.

### LINKED ISSUE #12: Placeholder permissions
Eventually we'll need to limit access to different endpoints to only users with the correct permission. Whether or not a user has permission will ultimately need to be determined via the DUNE RBAC system, however this is still currently undefined. As a placeholder we'll hide the views behind an arbitrary permission requirement but all users will have the permission by default.

### PR DESCRIPTION
**Pull request by jamesturner246**
# Description

Adds a set of permissions for modifying processes, and checks users for them when running views.

Tests are added for privileged vs unprivileged clients.

Fixes #12 

## Type of change

- [ ] Documentation (non-breaking change that adds or improves the documentation)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (non-breaking, back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (whatever its nature)

## Key checklist

- [x] All tests pass (eg. `python -m pytest`)
- [ ] The documentation builds and looks OK (eg. `python -m sphinx -b html docs docs/build`)
- [x] Pre-commit hooks run successfully (eg. `pre-commit run --all-files`)

## Further checks

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added or an issue has been opened to tackle that in the future. (Indicate issue here: # (issue))

### CODE REVIEW

File: main/models.py [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
+    class Meta:
+        """Meta class for the User model."""
+
+        permissions: ClassVar = [
+            ("can_boot_processes", "Can boot processes"),
+            ("can_restart_processes", "Can restart processes"),
+            ("can_flush_processes", "Can flush processes"),
+            ("can_kill_processes", "Can kill processes"),
+            ("can_view_process_logs", "Can view process logs"),
+        ]
```
Comment (alexdewar) [inline main/models.py:21]: Do we need all these separate permissions? I can see you *might* want some users to only be able to view the logs, but does it make sense to have users who can kill and boot processes, but not restart them?

**Review by alexdewar — CHANGES_REQUESTED** [review]
I'm not sure about having so many different permission types, but other than that it all seems good.

File: main/models.py (hunk 1 above)
Comment (jamesturner246) [reply main/models.py:21]: How about one for `view_logs` and one for `modify_processes`?

Comment (alexdewar) [reply main/models.py:21]: That makes sense to me: kind of a read vs write distinction.

**Review by alexdewar — APPROVED** [review]
LGTM :+1:

**Review by cc-a — CHANGES_REQUESTED** [review]
Implementation looks good. Just needs tests for the views.

**Review by dalonsoa — CHANGES_REQUESTED** [review]
Looks good, but I agree with Chris - we need tests to check that the views have the right `permission_required`.

**Comment by jamesturner246** [comment]
Okay, the test classes now also check for permissions. Ready to go.

**Review by cc-a — APPROVED** [review]
Grand. Looks good to me. I'll leave this up for people to look at for now and will probably merge it towards the end of the day.

**Review by dalonsoa — APPROVED** [review]
LGTM!

**Pull request merged**